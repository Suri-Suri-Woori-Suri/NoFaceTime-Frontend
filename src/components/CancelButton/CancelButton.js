import React from 'react';
import styles from './CancelButton.module.css';

const CancelButton = ({
  buttonName,
  onClick
}) => {
  const cancel = onClick;
  console.log("$$$$$$$", cancel);

  return (
    <button
      className={styles.CancelButton}
      onClick={cancel}>
      {buttonName}
    </button>
  );
};

export default CancelButton;
