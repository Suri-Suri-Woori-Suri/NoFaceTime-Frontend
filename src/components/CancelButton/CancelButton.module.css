.CancelButton {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  width: 100px;
  height: 40px;
  background: #7f0000;
  color: #ffffff;
  border-color: #7f0000;
  border-width: 1px;
  border-style: solid;
  border-radius: 20px 20px 20px 20px;
  font-family: "Oswald";
  font-weight: 500;
  font-size: 1rem;
  text-align: center;
}

.CancelButton:hover {
  transform: translateY(2px);
}
