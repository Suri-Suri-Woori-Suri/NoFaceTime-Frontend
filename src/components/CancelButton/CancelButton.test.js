import React from 'react';
import { mount } from 'enzyme';

import CancelButton from './CancelButton';

describe('<CancelButton/>', () => {
  const mockButtonName = '취소';
  const mockOnClick = jest.fn();

  const wrapper = mount(
    <CancelButton
      buttonName={mockButtonName}
      onClick={mockOnClick} />
  );

  it('should be contain button', () => {
    const button = wrapper.findWhere(node =>
      node.type() === 'button');

    expect(button.text()).toBe(mockButtonName);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
