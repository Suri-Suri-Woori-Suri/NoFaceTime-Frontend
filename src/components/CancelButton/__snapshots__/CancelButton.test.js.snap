// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CancelButton/> matches snapshot 1`] = `
<CancelButton
  buttonName="취소"
  onClick={[MockFunction]}
>
  <button
    className="CancelButton"
    onClick={[MockFunction]}
  >
    취소
  </button>
</CancelButton>
`;
